apps/designer/app/designer/features/style-panel/shared/style-info.ts

```diff
@@ -23,31 +57,164 @@ export const getStyleSource = (
   return "preset";
 };
 
+const styleProperties = Object.keys(properties) as StyleProperty[];
+const inheritableProperties = new Set<string>();
+for (const [property, value] of Object.entries(properties)) {
+  if (value.inherited) {
+    inheritableProperties.add(property);
+  }
+}
+
+/**
+ * find all breakpoints which may affect current view
+ */
+export const getCascadedBreakpointIds = (
+  breakpoints: Breakpoint[],
+  selectedBreakpointId?: string
+) => {
+  const sortedBreakpoints = breakpoints
+    .slice()
+    .sort((a, b) => a.minWidth - b.minWidth);
+  const cascadedBreakpointIds: string[] = [];
+  for (const breakpoint of sortedBreakpoints) {
+    if (breakpoint.id === selectedBreakpointId) {
+      break;
+    }
+    cascadedBreakpointIds.push(breakpoint.id);
+  }
+  return cascadedBreakpointIds;
+};
+
+/**
+ * extract all styles from breakpoints
+ * affecting current view
+ */
+export const getCascadedInfo = (
+  cssRules: CssRule[],
+  cascadedBreakpointIds: string[]
+) => {
+  const styles = new Map<string, Style>();
+  for (const rule of cssRules) {
+    if (rule.breakpoint !== undefined) {
+      styles.set(rule.breakpoint, rule.style);
+    }
+  }
+  const cascadedStyle: CascadedInfo = {};
+  for (const breakpointId of cascadedBreakpointIds) {
+    const style = styles.get(breakpointId);
+    if (style !== undefined) {
+      for (const [property, value] of Object.entries(style)) {
+        cascadedStyle[property as StyleProperty] = {
+          breakpointId,
+          value,
+        };
+      }
+    }
+  }
+  return cascadedStyle;
+};
+
+/**
+ * extract all inheritable styles from ancestor instances
+ * including active breakpoints
+ */
+export const getInheritedInfo = (
+  rootInstance: Instance,
+  instanceId: string,
+  cascadedAndSelectedBreakpoints: string[]
+) => {
+  const inheritedStyle: InheritedInfo = {};
+  const parents = utils.tree.getInstancePath(rootInstance, instanceId);
+  for (const parentInstance of parents) {
+    // skip current element
+    if (parentInstance.id === instanceId) {
+      continue;
+    }
+    const cascadedStyle = getCascadedInfo(
+      parentInstance.cssRules,
+      cascadedAndSelectedBreakpoints
+    );
+    for (const [property, cascaded] of Object.entries(cascadedStyle)) {
+      if (cascaded !== undefined && inheritableProperties.has(property)) {
+        inheritedStyle[property as StyleProperty] = {
```

Comment:
From the code above it looks like every inheritable property must be in cascadedStyle, but cascaded doesn't include the current breakpoint style. What if it's inherited from the current and not included in any cascaded?

Reply:
I got, `cascadedAndSelectedBreakpoints`

Reply:
Probably better to split as everywhere we see cascadedBreakpoints, but here not. May be pass selected as an argument and to merge it here in function? Anyway ok, just little bit confusing after previous functions ;-)